flink-yarn/src/main/java/org/apache/flink/yarn/Utils.java

```diff
@@ -620,4 +623,30 @@ public static YarnConfiguration getYarnConfiguration(
 
         return yarnConfig;
     }
+
+    /**
+     * Sets the application ACLs for the given ContainerLaunchContext based on the values specified
+     * in the given Flink configuration. Only ApplicationAccessType.VIEW_APP and
+     * ApplicationAccessType.MODIFY_APP ACLs are set, and only if they are configured in the Flink
+     * configuration.
+     *
+     * @param amContainer the ContainerLaunchContext to set the ACLs for
+     * @param flinkConfig the Flink configuration to read the ACL values from
+     */
+    public static void setAclsFor(
+            ContainerLaunchContext amContainer,
+            org.apache.flink.configuration.Configuration flinkConfig) {
+        Map<ApplicationAccessType, String> acls = new HashMap<>();
+        String viewAcls = flinkConfig.getString(YarnConfigOptions.APPLICATION_VIEW_ACLS, null);
+        String modifyAcls = flinkConfig.getString(YarnConfigOptions.APPLICATION_MODIFY_ACLS, null);
+        if (viewAcls != null) {
+            acls.put(ApplicationAccessType.VIEW_APP, viewAcls);
+        }
+        if (modifyAcls != null) {
+            acls.put(ApplicationAccessType.MODIFY_APP, modifyAcls);
+        }
```

Comment:
By default include `user.name` always ? (for view and modify acl)

Reply:
User who is running the Flink application will always have the right to view the logs and kill the application irrespective of the `yarn.view.acls` and `yarn.modify.acls` values.

Reply:
Looks like Spark is explicitly adding `user.name and SPARK_USER` both to the acls by default. Although from this [doc](https://docs.cloudera.com/runtime/7.2.7/yarn-security/yarn-security.pdf), looks like the `Users who start an application (the owners) always have access to the application they start, which includes the application logs, job statistics, and ACLs`